span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java

```diff
@@ -572,10 +570,21 @@ private static URL getNormalizedUrl(String url) throws MalformedURLException {
   private void maybeSetHttpUrlForOtelFormat(
       Request.Builder requestBuilder,
       final Map<String, AttributeValue> attributeValueMap) {
-    if (requestBuilder.hasUrl()) {
+    if (requestBuilder.hasUrl() && isAbsoluteUrl(requestBuilder.getUrl())) {
       return;
     }
+    // if requestBuilder.getUrl() is not absolute try building the url from other attributes
     Optional<String> url = HttpSemanticConventionUtils.getHttpUrlForOTelFormat(attributeValueMap);
     url.ifPresent(requestBuilder::setUrl);
   }
+
+  static boolean isAbsoluteUrl(String urlStr) {
```

Comment:
I believe it's the same method defined in EnrichedSpanUtils above.

Reply:
Yes it is, since `span-normalizer` doesn't depend on `enriched-span-constants` or vice-versa, I just duplicated the code.